File: CS/MeffedNavigationService.ModuleB/GroupsModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using MeffedNavigationService.Infrastructure.Contract;
using MeffedNavigationService.Infrastructure.DataModel;
using MeffedNavigationService.Infrastructure.Enums;

namespace MeffedNavigationService.ModuleB {
    public class GroupsModule : IModuleListing {
        List<MenuDataItem> _menuItems;
        /// <summary>
        /// Initializes a new instance of the GroupsModule class.
        /// </summary>
        /// <param name="menuItems"></param>
        public GroupsModule() {
            _menuItems = new List<MenuDataItem>();
            this.InitializeGroupsModule();
        }
        public List<MenuDataItem> MenuItems {
            get { return this._menuItems; }
        }
        private void InitializeGroupsModule() {
            MenuItems.Add(new MenuDataItem("Products", ModuleTypes.ViewProducts, "Opportunities", true));
            MenuItems.Add(new MenuDataItem("Orders", ModuleTypes.ViewOrders));
            MenuItems.Add(new MenuDataItem("Quotes", ModuleTypes.ViewQuotes));
            MenuItems.Add(new MenuDataItem("Invoices", ModuleTypes.ViewInvoices));
        }
        #region IModuleListing Members
        IEnumerable<MenuDataItem> IModuleListing.MenuItems {
            get { return this.MenuItems; }
        }
        #endregion
    }
}

File: VB/MeffedNavigationService.ModuleB/GroupsModule.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports MeffedNavigationService.Infrastructure.Contract
Imports MeffedNavigationService.Infrastructure.DataModel
Imports MeffedNavigationService.Infrastructure.Enums

Namespace MeffedNavigationService.ModuleB
	Public Class GroupsModule
		Implements IModuleListing
		Private _menuItems As List(Of MenuDataItem)
		''' <summary>
		''' Initializes a new instance of the GroupsModule class.
		''' </summary>
		''' <param name="menuItems"></param>
		Public Sub New()
			_menuItems = New List(Of MenuDataItem)()
			Me.InitializeGroupsModule()
		End Sub
		Public ReadOnly Property MenuItems() As List(Of MenuDataItem)
			Get
				Return Me._menuItems
			End Get
		End Property
		Private Sub InitializeGroupsModule()
			MenuItems.Add(New MenuDataItem("Products", ModuleTypes.ViewProducts, "Opportunities", True))
			MenuItems.Add(New MenuDataItem("Orders", ModuleTypes.ViewOrders))
			MenuItems.Add(New MenuDataItem("Quotes", ModuleTypes.ViewQuotes))
			MenuItems.Add(New MenuDataItem("Invoices", ModuleTypes.ViewInvoices))
		End Sub
		#Region "IModuleListing Members"
		Private ReadOnly Property IModuleListing_MenuItems() As IEnumerable(Of MenuDataItem) Implements IModuleListing.MenuItems
			Get
				Return Me.MenuItems
			End Get
		End Property
		#End Region
	End Class
End Namespace
